Operator SIGSEGV when RGW has TLS enabled

**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

When RGW has TLS certificate set , rook-operator pod crash

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1d32433]
goroutine 1221 [running]:
github.com/rook/rook/pkg/operator/ceph/object.(*bucketChecker).checkObjectStoreHealth(0xc002422000, 0xc00158f200, 0xc001a08fb0)
	/home/rook/go/src/github.com/rook/rook/pkg/operator/ceph/object/health.go:141 +0x1173
github.com/rook/rook/pkg/operator/ceph/object.(*bucketChecker).checkObjectStore(0xc002422000, 0xc00066b200)
	/home/rook/go/src/github.com/rook/rook/pkg/operator/ceph/object/health.go:82 +0x45
created by github.com/rook/rook/pkg/operator/ceph/object.(*ReconcileCephObjectStore).startMonitoring
	/home/rook/go/src/github.com/rook/rook/pkg/operator/ceph/object/controller.go:542 +0x3e5
```

**Expected behavior:**

rook-operator pod doesn't crash.

**How to reproduce it (minimal and precise):**

Create ObjectStore with TLS enable with a cert like that : 

```
-----BEGIN RSA PRIVATE KEY-----
XXX
-----END RSA PRIVATE KEY-----
-----BEGIN CERTIFICATE-----
XXX
-----END CERTIFICATE-----
-----BEGIN CERTIFICATE-----
XXX
-----END CERTIFICATE-----
```

This issue occur since this commit d86681244e940d34e37b4930820f89f49958b56e

**Environment**:
* OS (e.g. from /etc/os-release): Ubuntu 20.04 LTS
* Kernel (e.g. `uname -a`):5.4.0
* Cloud provider or hardware configuration: Bare metal
* Rook version (use `rook version` inside of a Rook Pod): v1.6.5
* Storage backend version (e.g. for ceph do `ceph -v`): v15.2.11
* Kubernetes version (use `kubectl version`):  v1.19.9
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): kubeadm
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)): HEALTH_OK

More information can be found on this Slack thread : https://rook-io.slack.com/archives/CK9CF5H2R/p1623839487498600

--- a/pkg/operator/ceph/object/bucket/provisioner.go
+++ b/pkg/operator/ceph/object/bucket/provisioner.go
@@ -22,7 +22,6 @@ import (
 	"net/http"
 	"strconv"
 	"strings"
-	"time"
 
 	"github.com/aws/aws-sdk-go/aws/awserr"
 	"github.com/ceph/go-ceph/rgw/admin"
@@ -622,7 +621,7 @@ func (p *Provisioner) setTlsCaCert() error {
 func (p *Provisioner) setAdminOpsAPIClient() error {
 	// Build TLS transport for the HTTP client if needed
 	httpClient := &http.Client{
-		Timeout: time.Second * 15,
+		Timeout: cephObject.HttpTimeOut,
 	}
 	if p.tlsCert != nil {
 		httpClient.Transport = cephObject.BuildTransportTLS(p.tlsCert)

--- a/pkg/operator/ceph/object/config.go
+++ b/pkg/operator/ceph/object/config.go
@@ -21,6 +21,7 @@ import (
 	"path"
 	"strconv"
 	"strings"
+	"time"
 
 	"github.com/pkg/errors"
 	cephconfig "github.com/rook/rook/pkg/operator/ceph/config"
@@ -42,6 +43,7 @@ caps osd = "allow rwx"
 	certKeyFileName                = "rgw-key.pem"
 	rgwPortInternalPort      int32 = 8080
 	ServiceServingCertCAFile       = "/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt"
+	HttpTimeOut                    = time.Second * 15
 )
 
 var (

--- a/pkg/operator/ceph/object/health.go
+++ b/pkg/operator/ceph/object/health.go
@@ -131,13 +131,15 @@ func (c *bucketChecker) checkObjectStoreHealth() error {
 	// Generate unique user and bucket name
 	bucketName := genUniqueBucketName(c.objContext.UID)
 	userConfig := c.genUserConfig()
-
-	var httpClient *http.Client
+	httpClient := &http.Client{
+		Timeout: HttpTimeOut,
+	}
 	if c.objectStoreSpec.IsTLSEnabled() {
 		tlsCert, err = GetTlsCaCert(c.objContext, c.objectStoreSpec)
 		if err != nil {
 			return errors.Wrapf(err, "failed to fetch CA cert for the user %q to establish TLS connection with the object store %q", userConfig.ID, c.namespacedName.Name)
 		}
+
 		httpClient.Transport = BuildTransportTLS(tlsCert)
 	}
 

--- a/pkg/operator/ceph/object/s3-handlers.go
+++ b/pkg/operator/ceph/object/s3-handlers.go
@@ -22,7 +22,6 @@ import (
 	"crypto/x509"
 	"net/http"
 	"strings"
-	"time"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/aws/awserr"
@@ -45,7 +44,7 @@ func NewS3Agent(accessKey, secretKey, endpoint string, debug bool, tlsCert []byt
 		logLevel = aws.LogDebug
 	}
 	client := http.Client{
-		Timeout: time.Second * 15,
+		Timeout: HttpTimeOut,
 	}
 	tlsEnabled := false
 	if len(tlsCert) > 0 {

--- a/pkg/operator/ceph/object/user/controller.go
+++ b/pkg/operator/ceph/object/user/controller.go
@@ -333,7 +333,9 @@ func (r *ReconcileObjectStoreUser) initializeObjectStoreContext(u *cephv1.CephOb
 	}
 
 	// Build TLS client if needed
-	var httpClient *http.Client
+	httpClient := &http.Client{
+		Timeout: object.HttpTimeOut,
+	}
 	if store.Spec.IsTLSEnabled() {
 		tlsCert, err := object.GetTlsCaCert(objContext, &store.Spec)
 		if err != nil {

